PREOPS-3596: DECam Effective Exposure Time Calculation

kadrlica opened the pull request:
A pedagogical investigation of the DES/DESDM effective exposure time calculation, and a translation/replication using the LSST Science Pipelines. A step in the direction of implementing the effective exposure time metric in analysis_tools.

ctslater reviewed (approved):
This is great; I learned a lot from reading it and I think we'll be glad to have this to refer back to in the future. This probably can form the basis for a technote once the metric is in operation. It also sounds like it will be useful to publish the three separate factors as their own metrics in addition to the combined value.

Most minor of comments:
"representing the independent affects" -> "effects". "Effictive" in one of the section headings. 
I'm wary of patterns like `butler = dafButler.Butler(repo, collections=allcollections)`, we don't really know where this data is coming from if it's initialized like that. (I'm not sure what problem is solved by doing this?)

kadrlica commented:
Thanks @ctslater! I've fixed the typos.

Sorry about the "allcollections" pattern. This was left over from the exploration phase when I was trying to find the right data sets to use. I've replaced it with a simple `butler = dafButler.Butler("/repo/main")`. Later on `queryDatasets` is grabbing from a specific collection that is explicitly specified.
